src/qat/purr/compiler/builders.py

```diff
@@ -133,7 +143,7 @@ def _fix_clashing_label_names(
 
             inst.name = new_name
 
-    def merge_builder(self, other_builder: "InstructionBuilder"):
+    def merge_builder(self, other_builder: "InstructionBuilder", index: int = None):
```

Comment:
`index` here is ambiguous to the caller, can you propose a better name ?

Reply:
I'll add a docstring explanation. Index makes sense to me so unless you have another suggestion I don't know what I would replace it with.